### PR TITLE
Move handlers so they are lazy loaded

### LINKED ISSUE #5: Question: Is hot reload supported?
It would be awesome if hot reloading was supported in the Hapi server. Do you have any plans to support that?

### LINKED ISSUE #6: Question: Is it possible to use function environment variables?
In my current scenario I have two lambdas. Lambda 1 invokes lambda 2. Serverless offline is loading lambda 1 and sets the environment variables based on the `serverless.yml` file config. Lambda 2 also has environment variables set; however, `process.env` doesn't contain them when lambda 2 is invoked. Lambda 2 doesn't have access to the environment variables set by serverless offline either (example `IS_OFFLINE`). 

Do you think it is possible to load the environment variables from the function and inject IS_OFFLINE when the local Hapi server is spun up? 

Thank you for the great plugin 🎉

### PR DESCRIPTION
**Pull request by tclindner**
Hey @marconi1992 - This PR moves the `handlers` so they are lazy loaded (as discussed in #6). Please let me know if you have any questions. 

Closes #5
Closes #6

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -43,6 +36,14 @@ class LambdaOffline {
           const invocationType = req.headers['x-amz-invocation-type'];
           const { functionName } = req.params;
 
+          const handlers = Object.keys(this.service.functions).reduce((acc, key) => {
+            const fun = this.service.getFunction(key);
+            const funOptions = functionHelper.getFunctionOptions(fun, key, servicePath, serviceRuntime);
```
Comment (marconi1992) [inline src/index.js]: I think the function options should be only loaded when serverless offline runs the first time, and also we should create the handler when the lambda function needs to be invoked to hot reload it.

outside the route
```js
const funOptions = Object.keys(this.service.functions).reduce((acc, key) => {
      const fun = this.service.getFunction(key);
      acc[key] = functionHelper.getFunctionOptions(fun, key, servicePath, serviceRuntime);
      return acc;
    }, {});
```

into the route

```js
const options = funOptions[functionName];

if (!options) {
   return reply().code(404);
}

const handler = functionHelper.createHandler(options, {});
```

**Review by marconi1992 — CHANGES_REQUESTED** [review]
It works!! thank you!! we can improve it to avoid load the function options every time the lambda is invoked.

File: src/index.js (hunk 1 above)
Comment (tclindner) [reply src/index.js]: Sounds good to me 👍 Thank you for looking through the PR!

File: src/index.js [hunk 2]
```diff
@@ -36,13 +42,13 @@ class LambdaOffline {
...
-          }, {});
+          const options = funOptions[functionName];
+
+          if (!options) {
+            return reply().code(404);
+          }
+
+          const handler = functionHelper.createHandler(options, {});
 
           const handler = handlers[functionName];
```
Comment (marconi1992) [inline src/index.js]: This `handler` constant is not necessary anymore, we should remove it

**Review by marconi1992 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (tclindner) [reply src/index.js]: Updated. Would you be open to adding a linter to help detect things like that?

Comment (marconi1992) [reply src/index.js]: We already have an ESlint configuration and npm script to run it: https://github.com/marconi1992/serverless-offline-lambda/blob/master/package.json#L11

I'd like to add a CI/CD. I'll create an issue to address that.

**Review by marconi1992 — APPROVED** [review]

**Pull request merged**